src/components/Main/Articles/ArticlesListedByTopic.jsx

```diff
@@ -5,28 +5,69 @@ import ArticleCard from './ArticleCard';
 import './Article.css';
 import Loading from '../../UI/Loading';
 import Error from '../../UI/error';
-import { useParams } from 'react-router-dom';
-
+import { useParams , useSearchParams } from 'react-router-dom';
+import Topics from './Topics';
 
 function ListArticlesByTopic() {
-    const {topic} = useParams()
-   const [articles, setArticles] = useState([]);
+    const { topic } = useParams();
+  const [articles, setArticles] = useState([]);
   const [error, setError] = useState(null);
+  const [sortCriteria, setSortCriteria] = useState('created_at');
+  const [sortOrder, setSortOrder] = useState('desc');
+  const [searchParams, setSearchParams] = useSearchParams();
 
   useEffect(() => {
-  
-    axios.get(`https://nc-news-vvdv.onrender.com/api/articles?topic=${topic}`)
+    const sort_by = searchParams.get('sort_by') || 'created_at';
+    const order = searchParams.get('order') || 'desc';
+
+    axios.get(`https://nc-news-vvdv.onrender.com/api/articles`, {
+      params: {
+        topic,
+        sort_by,
+        order,
+      },
```

Comment:
Rather than have a separate articlesByTopic, it is possible to make your articles component dynamic so it works for all articles or articles by topic